extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -230,6 +229,13 @@ public QueryDefinition makeQueryDefinition(
             partitionColumnNames
         );
 
+        // We need to pass only the window factories for the window stage definition.
+        // Sorting and partitioning are expected to be handled by the shuffle spec of the previous stage.
+        final List<OperatorFactory> windowOperatorList = operatorList.get(i)
+                                                                     .stream()
+                                                                     .filter(operator -> operator instanceof WindowOperatorFactory)
```

Comment:
this is not necessarily true - I don't believe in building an infrastrucutre which could only do partitioning/shuffle in-between stages

there are benefits of further partitioning an existing sort; and that's rather cheap....

Reply:
Currently, we are passing incorrect shuffle spec, and then explicitly passing sort+partition operators. So we're doing unnecessary additional work.
This PR fixes the shuffle spec, and removes the sort+partition operators.

> there are benefits of further partitioning an existing sort; and that's rather cheap....

Right now, MSQ creates the stages based on the list of operator factories we get from `Windowing.java` layer. So logic chance for optimisation like this should happen there IIUC? So it feels unrelated to this PR to me.